src/core/core.plugin.js

```diff
@@ -1,42 +1,102 @@
 "use strict";
 
 module.exports = function(Chart) {
-	var helpers = Chart.helpers;
-
-	// Plugins are stored here
-	Chart.plugins = [];
-	Chart.pluginService = {
-		// Register a new plugin
-		register: function(plugin) {
-			var p = Chart.plugins;
-			if (p.indexOf(plugin) === -1) {
-				p.push(plugin);
-			}
+
+	var noop = Chart.helpers.noop;
+
+	/**
+	 * The plugin service singleton
+	 * @namespace Chart.plugins
+	 * @since 2.1.0
+	 */
+	Chart.plugins = {
+		_plugins: [],
+
+		/**
+		 * Registers the given plugin(s) if not already registered.
+		 * @param {Array|Object} plugins plugin instance(s).
+		 */
+		register: function(plugins) {
+			var p = this._plugins;
+			([]).concat(plugins).forEach(function(plugin) {
+				if (p.indexOf(plugin) === -1) {
+					p.push(plugin);
+				}
+			});
 		},
 
-		// Remove a registered plugin
-		remove: function(plugin) {
-			var p = Chart.plugins;
-			var idx = p.indexOf(plugin);
-			if (idx !== -1) {
-				p.splice(idx, 1);
-			}
+		/**
+		 * Unregisters the given plugin(s) only if registered.
+		 * @param {Array|Object} plugins plugin instance(s).
+		 */
+		unregister: function(plugins) {
```

Comment:
I thought the same first but then realized that this method has never been used in code or plugins (correct me if I'm wrong), and I can't see any case where it should be used (except in unit tests). So was thinking that we may not keep compatibility and save a few bits. What do you think?